postgres/pgdump.go

```diff
@@ -185,13 +185,49 @@ func processStatements(conv *internal.Conv, statements []nodes.Node) *copyOrInse
 			if conv.SchemaMode() {
 				processVariableSetStmt(conv, n)
 			}
+		case nodes.IndexStmt:
+			if conv.SchemaMode() {
+				processIndexStmt(conv, n)
+			}
 		default:
 			conv.SkipStatement(prNodes([]nodes.Node{node}))
 		}
 	}
 	return nil
 }
 
+func toSchemaIndexKeys(indexParams nodes.List) (keys []schema.Key) {
+	var schemaKeys []schema.Key
+	for _, ip := range indexParams.Items {
+		schemaKeys = append(schemaKeys, schema.Key{Column: *ip.(nodes.IndexElem).Name})
+	}
+	return schemaKeys
+}
+
+func processIndexStmt(conv *internal.Conv, n nodes.IndexStmt) {
+	if n.Relation == nil {
+		logStmtError(conv, n, fmt.Errorf("cannot process index statement with nil relation."))
+		return
+	}
+	tableName, err := getTableName(conv, *n.Relation)
+	if err != nil {
+		logStmtError(conv, n, fmt.Errorf("can't get table name: %w", err))
+		return
+	}
+	if _, ok := conv.SrcSchema[tableName]; ok {
+		ctable := conv.SrcSchema[tableName]
+		ctable.Indexes = append(ctable.Indexes, schema.Index{
+			Name:   *n.Idxname,
+			Unique: *&n.Unique,
```

Comment:
*&n.Unique -- isn't this just the same as n.Unique?

Reply:
yes its same, updated to n.Unique